[umi-types] webpack-chain模块的类型定义有问题

## What happens?

在开发umi插件时，`webpack-chain`模块是使用`export =`导出类型的，`umi-types`在引入`webpack-chain`模块时，是
```javascript
import IWebpackChainConfig from 'webpack-chain';
```
这样引入的，这样会报错误` error TS1259: Module '"node_modules/webpack-chain/types/index"' can only be default-imported using the 'esModuleInterop' flag`，可以使用`tsc --esModuleInterop`来解决。但是`--esModuleInterop`会导致其他的js模块和同样使用`export =`导出类型的模块出现问题。

建议修改成：
```javascript
import * as IWebpackChainConfig from 'webpack-chain';
```

## 最小可复现仓库

[https://github.com/duan602728596/umi-types-bug-example](https://github.com/duan602728596/umi-types-bug-example)

## 复现步骤，错误日志以及相关配置

1. `tsc`编译代码，
  会报错` This module is declared with using 'export =', and can only be used with a default import when using the 'esModuleInterop' flag`
2. `tsc --esModuleInterop`，编译成功，
  但是代码内使用`import * as module from 'module-name';`方式导入的模块会报错` Type originates at this import. A namespace-style import cannot be called or constructed, and will cause a failure at runtime. Consider using a default import or import require here instead.`。
3. `umi-types`改成`import * as IWebpackChainConfig from 'webpack-chain';`后，
  正常编译代码成功。

## 相关环境信息

- **Umi 版本**：umi-types v0.3.11
- **Node 版本**：v12.4.0
- **操作系统**：win10

--- a/packages/umi-build-dev/src/getWebpackConfig.ts
+++ b/packages/umi-build-dev/src/getWebpackConfig.ts
@@ -8,7 +8,7 @@ import nodeExternals from 'webpack-node-externals';
 const debug = require('debug')('umi-build-dev:getWebpackConfig');
 
 interface IOpts {
-  ssr?: boolean | IExportSSROpts;
+  ssr?: IExportSSROpts;
 }
 
 export default function(service: IApi, opts: IOpts = {}) {

--- a/packages/umi-types/config.d.ts
+++ b/packages/umi-types/config.d.ts
@@ -1,6 +1,6 @@
 // https://umijs.org/config/
 import { ExternalsElement, Condition } from 'webpack';
-import IWebpackChainConfig from 'webpack-chain';
+import * as IWebpackChainConfig from 'webpack-chain';
 import { IChangeWebpackConfigFunc } from './index';
 
 export type IPlugin<T = any> = string | [string, T];
@@ -66,9 +66,11 @@ export interface IAFWebpackConfig {
 }
 
 type WhitelistOption = string | RegExp;
-export interface IExportSSROpts {
-  externalWhitelist?: WhitelistOption[];
-}
+export type IExportSSROpts =
+  | {
+      externalWhitelist?: WhitelistOption[];
+    }
+  | boolean;
 
 interface IConfig extends IAFWebpackConfig {
   // basic config
@@ -92,7 +94,7 @@ interface IConfig extends IAFWebpackConfig {
   targets?: {
     [key: string]: number;
   };
-  ssr?: boolean | IExportSSROpts;
+  ssr?: IExportSSROpts;
 }
 
 export default IConfig;

--- a/packages/umi-types/config.d.ts
+++ b/packages/umi-types/config.d.ts
@@ -66,11 +66,9 @@ export interface IAFWebpackConfig {
 }
 
 type WhitelistOption = string | RegExp;
-export type IExportSSROpts =
-  | {
-      externalWhitelist?: WhitelistOption[];
-    }
-  | boolean;
+export type IExportSSROpts = {
+  externalWhitelist?: WhitelistOption[];
+} | boolean;
 
 interface IConfig extends IAFWebpackConfig {
   // basic config

--- a/packages/umi-types/index.d.ts
+++ b/packages/umi-types/index.d.ts
@@ -1,7 +1,7 @@
 import 'cheerio';
 import IConfig, { IPlugin, IAFWebpackConfig, IRoute } from './config';
 import { Stats, Configuration } from 'webpack';
-import IWebpackChainConfig from 'webpack-chain';
+import * as IWebpackChainConfig from 'webpack-chain';
 
 /**
  * System level variable

